src/components/ServiceList.jsx

```diff
@@ -1,22 +1,57 @@
 import { Alert } from "@baltimorecounty/dotgov-components";
 import FilterList from "../components/FilterList";
-import { MostPopularServiceIconStyles } from "../styles";
-import React from "react";
-import ServiceIconLink from "../components/ServiceIconLink";
+import React, { useState } from "react";
+import ServiceIconLink from "./ServiceIconLink";
+import PopularityFilterCollapse from "./PopularityFilterCollapse";
 import useServices from "../hooks/useServices";
+import { MostPopularServiceIconStyles } from "../styles";
+import { TextInput } from "@baltimorecounty/dotgov-components";
 
 const ServiceList = () => {
   const { hasError, serviceItems = [], isLoading } = useServices();
+  const [isMostPopular, setMostPopular] = useState(false);
+  const [filteredItems, setFilteredItems] = useState([]);
+  const [filterText, setFilterText] = useState([]);
+  const [isFiltering, setIsFiltering] = useState(false);
 
-  let legendText = {
-    fontStyle: "italic",
-    marginLeft: "5px",
-    fontWeight: "900"
+  const filterItems = (services, searchText) => {
+    return services.filter(item => {
+      const { name, department } = item;
+      return (
+        name.toLowerCase().indexOf(searchText.toLowerCase()) !== -1 ||
+        department.toLowerCase().indexOf(searchText.toLowerCase()) !== -1
+      );
+    });
   };
 
-  let flex = {
-    display: "flex",
-    flexDirection: "row"
+  const filterByPopularity = item => item.rank > 0;
+
+  const checkCondition = (checkedVal, searchText) => {
+    let checkedItems = [];
+    const searchTextLength = searchText.length > 0;
+    checkedItems = checkedVal
+      ? searchTextLength
+        ? filterItems(serviceItems.filter(filterByPopularity), searchText)
+        : serviceItems.filter(filterByPopularity)
+      : searchTextLength
+      ? filterItems(serviceItems, searchText)
+      : [];
+
+    setFilteredItems(checkedItems);
+  };
+
+  const handleIsPopularFilterChange = item => {
+    const checkedValue = item.target.checked;
+    setMostPopular(checkedValue);
+    setIsFiltering(filterText.length > 0 || checkedValue);
+    checkCondition(checkedValue, filterText);
+  };
+
+  const handleTextInputFilterChange = event => {
```

Comment:
same comment as here, https://github.com/baltimorecounty/react-services-list/pull/10/files#r369719104

Reply:
obviously it would be

```js
const { value } ...
```

Reply:
updated